File: Reverse Linked List/Reverse_Linked_List.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun reverseList(head: ListNode?): ListNode? {
        var current = head
        var next: ListNode? = null
        var prev: ListNode? = null
        while ( current != null ){
            next = current.next
            current.next = prev
            prev = current
            current = next
        }
        return prev
    }
}

File: Coding Problems & My Solutions/Reverse Linked List/Reverse_Linked_List.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init() { self.val = 0; self.next = nil; }
 *     public init(_ val: Int) { self.val = val; self.next = nil; }
 *     public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
 * }
 */
class Solution {
    func reverseList(_ head: ListNode?) -> ListNode? {
        var p: ListNode? = nil
        var n: ListNode? = nil
        var c = head
        while c != nil {
            n = c?.next
            c?.next = p
            p = c
            c = n
        }
        return p
    }
}